### PR TITLE
Episode 4 - A New Hope

### PR DESCRIPTION
**Pull request by TylorS**
# WIP

Just opening this for visibility 😄 

I'm thinking of first accessing the list of people, creating a simple visual grid of them, allowing the selection of one of them and then fetching the additional information available about them and displaying that. It's sorta stubbed out but not quite there

### CODE REVIEW

File: src/4-episode-taskeither/index.tsx [hunk 1]
```diff
@@ -46,7 +45,24 @@ export function useTaskEither<E, A>() {
...
+        value,
+        O.matchW(
+          () => (isLoading ? onLoading() : onNone()),
+          E.matchW(onError, onSuccess),
+        ),
+      ),
+    [],
+  )
+
+  return [match, run] as const
```
Comment (monsieurBoutte) [inline src/4-episode-taskeither/index.tsx]: love this pattern

File: src/4-episode-taskeither/index.tsx [hunk 2]
```diff
@@ -46,7 +45,24 @@ export function useTaskEither<E, A>() {
...
+  const match = React.useCallback(
+    <B, C, D, F>(
+      onNone: () => B,
+      onLoading: () => C,
+      onError: (error: E) => D,
+      onSuccess: (value: A) => F,
+    ) =>
+      pipe(
+        value,
+        O.matchW(
```
Comment (monsieurBoutte) [inline src/4-episode-taskeither/index.tsx:276]: 💭  what is `matchW` doing here, I assume widening the possibilities within the error channel yeah?

Comment (TylorS) [reply src/4-episode-taskeither/index.tsx:276]: It's actually widening the return value in this case

**Review by monsieurBoutte — APPROVED** [review]
Yeah man, this is solid af. I'll come in and tweak the styling a little bit but there's no need to tweak the logic 🎉

**Pull request merged**